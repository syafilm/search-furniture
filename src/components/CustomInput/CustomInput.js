/*eslint-disable */
import React from 'react'

import {
  Styles,
  Helpers,
} from 'utils'

const CustomInput = ({
  value,
  onBlur,
  onKeyUp,
  onFocus,
  type,
  placeholder,
  className,
}) => (
  <div className={Styles.positionRelative}>
    {
      type === 'textarea'
        ? (
          <textarea
            rows="4"
            defaultValue={value}
            className={className ? Helpers.mergeCss(Styles.input(type), Styles.formControl, className) : Helpers.mergeCss(Styles.input(type), Styles.formControl)}
            placeholder={placeholder}
            onBlur={onBlur}
            onKeyUp={onKeyUp}
            onFocus={onFocus}
          />
        )
        : (
          <input
            type={type}
            defaultValue={value}
            className={className ? Helpers.mergeCss(Styles.input(type), Styles.formControl, className) : Helpers.mergeCss(Styles.input(type), Styles.formControl)}
            placeholder={placeholder}
            onBlur={onBlur}
            onKeyUp={onKeyUp}
            onFocus={onFocus}
          />
        )
    }
  </div>
)

export default CustomInput
