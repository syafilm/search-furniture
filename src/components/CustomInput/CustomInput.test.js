import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import CustomInput from './CustomInput'

describe('Component CustomInput', () => {
  const mockEvent = jest.fn()
  const valueSearch = 'Ryan ryenold'
  const placeholder = 'Search Giphy here'
  it('Can render CustomInput with type component textarea', () => {
    const component = shallow(
      <CustomInput
        type="textarea"
        value={valueSearch}
        onKeyUp={mockEvent}
        onBlur={mockEvent}
        placeholder={placeholder}
      />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })

  it('Can render CustomInput even without placeholder', () => {
    const component = shallow(
      <CustomInput
        type="text"
        value={valueSearch}
        onKeyUp={mockEvent}
        onBlur={mockEvent}
      />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })

  it('Can render CustomInput default with type component text', () => {
    const component = shallow(
      <CustomInput
        value={valueSearch}
        onKeyUp={mockEvent}
        onBlur={mockEvent}
      />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })
})
