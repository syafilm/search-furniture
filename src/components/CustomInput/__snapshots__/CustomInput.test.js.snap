// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CustomInput Can render CustomInput default with type component text 1`] = `
<div
  className="css-cssveg"
>
  <input
    className="css-1why0iz"
    defaultValue="Ryan ryenold"
    onBlur={[MockFunction]}
    onKeyUp={[MockFunction]}
  />
</div>
`;

exports[`Component CustomInput Can render CustomInput even without placeholder 1`] = `
<div
  className="css-cssveg"
>
  <input
    className="css-1why0iz"
    defaultValue="Ryan ryenold"
    onBlur={[MockFunction]}
    onKeyUp={[MockFunction]}
    type="text"
  />
</div>
`;

exports[`Component CustomInput Can render CustomInput with type component textarea 1`] = `
<div
  className="css-cssveg"
>
  <textarea
    className="css-1f9izye"
    defaultValue="Ryan ryenold"
    onBlur={[MockFunction]}
    onKeyUp={[MockFunction]}
    placeholder="Search Furniture here"
    rows="4"
  />
</div>
`;
